File: backend/app/controllers/events.py
from sqlalchemy.orm import Session
from app.models.events import Event
from app.schemas.events import EventCreate, EventUpdate


# function for getting all events
def get_events(db: Session):
    return db.query(Event).all()


# function for creating a new event
def create_event(db: Session, event: EventCreate):
    db_event = Event(
        title=event.title,
        description=event.description,
        start_date=event.start_date,
        img_url=event.img_url
    )
    db.add(db_event)
    db.commit()
    db.refresh(db_event)
    return db_event


# function for updating an existing event by id
def update_event(db: Session, event_id: int, event: EventUpdate):
    db_event = db.query(Event).filter(Event.event_id == event_id).first()
    db_event.title = event.title
    db_event.description = event.description
    db_event.start_date = event.start_date
    db_event.img_url = event.img_url
    db.commit()
    db.refresh(db_event)
    return db_event


# function for deleting an existing event by id
def delete_event(db: Session, event_id: int):
    event = db.query(Event).filter(Event.event_id == event_id).first()
    db.delete(event)
    db.commit()
    return event

File: front/src/store/events.js
import instance from "@/middlewares";


export default {
  name: "events",
  state: () => ({
    data: null,

  }),
  mutations: {
    setData(state, data) {
      state.data = data;
    },

  },
  actions: {
    // class Event(Base):
    // __tablename__ = "events"

    // event_id = Column(BigInteger, primary_key=True, index=True)
    // title = Column(Text, nullable=False)
    // description = Column(Text, nullable=False)
    // start_date = Column(Date, nullable=False)
    // img_url = Column(Text, nullable=True)

    // events crud
    // get events
    async getEvents({ commit }) {
        try {
            const response = await instance.get("/api/events");
            if (response) return commit("setData", response.data);
        }
        catch (error) {
            console.log(error);
        }
    },

    // create event
    // title: string
    // description: string
    // start_date: string
    // img_url: string
    async createEvent({}, input) {
        try {
            const { title, description, start_date, img_url } = input;
            const response = await instance.post("/api/events", { title, description, start_date, img_url });
            if (response.ok) return console.log("ok");
        }
        catch (error) {
            console.log(error);
        }
    },

    // update event
    // title: string
    // description: string
    // start_date: string
    // img_url: string
    async updateEvent({}, input) {
        try {
            const { id, title, description, start_date, img_url } = input;
            const response = await instance.put(`/api/events/${id}`, { title, description, start_date, img_url });
            if (response.ok) return console.log("ok");
        }
        catch (error) {
            console.log(error);
        }
    },

    // delete event
    async deleteEvent({}, id) {
        try {
            const response = await instance.delete(`/api/events/${id}`);
            if (response.ok) return console.log("ok");
        }
        catch (error) {
            console.log(error);
        }
    },
    

},

namespaced: true,
};
